app/services/dark-mode.ts

```diff
@@ -0,0 +1,138 @@
+import Service, { service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { registerDestructor } from '@ember/destroyable';
+import type RouterService from '@ember/routing/router-service';
+import type RouteInfo from '@ember/routing/route-info';
+import type LocalStorageService from 'codecrafters-frontend/services/local-storage';
+import RouteInfoMetadata, { RouteColorSchemes } from 'codecrafters-frontend/utils/route-info-metadata';
+
+const LOCAL_STORAGE_KEY = 'dark-mode-preference';
+
+export type LocalStoragePreference = 'system' | 'dark' | 'light' | null;
+export type SystemPreference = 'dark' | 'light';
+
+export default class DarkModeService extends Service {
+  @service declare router: RouterService;
+  @service declare localStorage: LocalStorageService;
+
+  /**
+   * Currently loaded localStorage preference value
+   */
+  @tracked localStoragePreference?: LocalStoragePreference;
+
+  /**
+   * Currently loaded system preference value
+   */
+  @tracked systemPreference?: SystemPreference;
+
+  constructor(properties: object | undefined) {
+    super(properties);
+
+    // Load the localStorage preference from localStorage service
+    this.localStoragePreference = this.localStorage.getItem(LOCAL_STORAGE_KEY) as LocalStoragePreference;
```

Comment:
```suggestion
    this.localStoragePreference = this.localStorage.getItem(LOCAL_STORAGE_KEY) || undefined;
```

(Using `as xyz` could result in `localStoragePreference` being set to `null`, which doesn't agree w/ the types)

Reply:
The type does allow `null` though:

![image](https://github.com/codecrafters-io/frontend/assets/493875/e5663494-bdc1-42da-861d-a01186580bc0)

Reply:
Ah, ignore me - I thought I saw a `localStoragePreference?` somewhere. All good!